vscode/src/azure/treeView.ts

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as vscode from "vscode";
+import { queryWorkspace } from "./workspaceActions";
+import { log } from "qsharp-lang";
+
+// See docs at https://code.visualstudio.com/api/extension-guides/tree-view
+
+// Convert a date such as "2023-07-24T17:25:09.1309979Z" into local time
+function localDate(date: string) {
+  return new Date(date).toLocaleString();
+}
+
+export class WorkspaceTreeProvider
+  implements vscode.TreeDataProvider<WorkspaceTreeItem>
+{
+  private treeState: Map<string, WorkspaceConnection> = new Map();
+
+  private _onDidChangeData = new vscode.EventEmitter<
+    WorkspaceTreeItem | undefined
+  >();
```

Comment:
nit: why are private members using different naming conventions?

Reply:
I was just following the docs on this one (https://code.visualstudio.com/api/extension-guides/tree-view#updating-tree-view-content). I can rename if need-be.